content/guides/guides/web-security.md

```diff
@@ -0,0 +1,485 @@
+policies do not match.
+
+Because Cypress works from within the browser, Cypress must be able to directly
+communicate with your remote application at all times. Unfortunately, browsers
+naturally try to prevent Cypress from doing this.
+
+To get around these restrictions, Cypress implements some strategies involving
+JavaScript code, the browser's internal APIs, and network proxying to _play by
+the rules_ of same-origin policy. It is our goal to fully automate the
+application under test without you needing to modify your application's code -
+and we are _mostly_ able to do this.
+
+#### Examples of what Cypress does under the hood:
+
+- Injects
+  [`document.domain`](https://developer.mozilla.org/en-US/docs/Web/API/Document/domain)
+  into `text/html` pages.
+- Proxies all HTTP / HTTPS traffic.
+- Changes the hosted URL to match that of the application under test.
+- Uses the browser's internal APIs for network level traffic.
+
+When Cypress first loads, the internal Cypress web application is hosted on a
+random port: something like `http://localhost:65874/__/`.
+
+After the first [`cy.visit()`](/api/commands/visit) command is issued in a test,
+Cypress changes its URL to match the origin of your remote application, thereby
+solving the first major hurdle of same-origin policy. Your application's code
+executes the same as it does outside of Cypress, and everything works as
+expected.
+
+<Alert type="info">
+
+<strong class="alert-header">How is HTTPS supported?</strong>
+
+Cypress does some pretty interesting things under the hood to make testing HTTPS
+sites work. Cypress enables you to control and stub at the network level.
+Therefore, Cypress must assign and manage browser certificates to be able to
+modify the traffic in real time.
+
+You'll notice Chrome display a warning that the 'SSL certificate does not
+match'. This is normal and correct. Under the hood we act as our own CA
+authority and issue certificates dynamically in order to intercept requests
+otherwise impossible to access. We only do this for the superdomain currently
+under test, and bypass other traffic. That's why if you open a tab in Cypress to
+another host, the certificates match as expected.
+
+Note, that Cypress allows you to optionally specify CA / client certificate
+information for use with HTTPS sites. See
+[Configuring client certificates](/guides/references/client-certificates). If
+the remote server requests a client certificate for a configured URL, Cypress
+will supply it.
+
+</Alert>
+
+## Limitations
+
+It's important to note that although we do our **very best** to ensure your
+application works normally inside of Cypress, there _are_ some limitations you
+need to be aware of.
+
+### Different superdomain per test requires `cy.origin` command
+
+Cypress changes its own host URL to match that of your applications. With the
+exception of `cy.origin`, Cypress requires that the URLs navigated to have the
+same superdomain for the entirety of a single test.
```

Comment:
I think the word "superdomain" is a new one for a lot of readers so whenever it appears for the first time on a page I would like to see it linked to the explainer you added in `cross-origin-testing`

Reply:
maybe linking `/guides/guides/cross-origin-testing#Parts-of-a-URL`?